src/wallet/wallet2.cpp

```diff
@@ -5035,27 +5035,27 @@ std::string wallet2::make_uri(const std::string &address, const std::string &pay
   }
 
   std::string uri = "monero:" + address;
-  bool n_fields = 0;
+  int n_fields = 0;
 
   if (!payment_id.empty())
   {
-    uri += (n_fields++ ? "&" : "?") + std::string("tx_payment_id=") + payment_id;
+    uri += (n_fields++ > 0 ? "&" : "?") + std::string("tx_payment_id=") + payment_id;
```

Comment:
With the change from `bool` to `int`, the explicit condition was added for clarity, since the question is whether or not it's the first parameter. I removed the comparison and made the counter unsigned.